### ISSUE TITLE
Error shapes not alligned when creating Instance with numpy array

### ISSUE THREAD

**Issue by KamodaP** [jorgenkg/python-neural-network#19]
It is quite common in data science apps to use numpy arrays and pandas dataframes. Unfortunately when you try to create Instance using numpy array of shape (1,x) your code creates in one of the places array of shape (nobs, x, 1) which does not work in dot when multiplied by array of shape (x, nlayers). It could be easily checked with the verify_dataset_shape_and_modify. Here's some snippet:
```python
def mult(x):
     res = x[0]
     for a in x[1:]:
         res *= a
     return res
[...]
def verify_dataset_shape_and_modify(...)
    [...]
    data              = np.array( [instance.features.reshape((mult(instance.features.shape),) \
                                         if type(instance.features) == np.ndarray\ #more generalization perhaps?
                                         else instance.features\
                                         for instance in dataset ] )
```
whith this snippet even accidentally passing arrays of shape (x,1) will work. Or you can add `if a != 1: throw ValueError('Invalid shape')`

**Comment by jorgenkg** [comment]
You are correct. The documentation states that an instance should be initialized with a built-in one-dimensional python list, e.g. `Instance( [0, 1] )`. However, as you pint out there are currently no checks that verify whether the argument is indeed a python list. 

I'll add validation check to my todo list, thank you for pointing it out
